src/vcf/debugulator.cpp

```diff
@@ -0,0 +1,67 @@
+/**
+ * Copyright 2016 EMBL - European Bioinformatics Institute
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "vcf/debugulator.hpp"
+
+namespace ebi
+{
+  namespace vcf
+  {
+    namespace debugulator
+    {
+      bool fix_vcf_file(std::istream &input,
+                        ebi::vcf::ReportReader &errorDAO,
+                        std::ostream &output)
+      {
+          std::vector<char> line;
+          line.reserve(default_line_buffer_size);
+
+          size_t current_line = 0;  // the first line is the number 1, ParsingState takes this convention too
+
+          size_t errors = errorDAO.count_errors();
+          if (errors == 0) {
+              throw std::runtime_error{"The errors DB was empty, there are no errors to fix the input"};
```

Comment:
I don't think this should raise an error, an empty DB will be the output for any correct VCF file so it's not an erroneous situation. This message could be simply displayed.